src/main/python/resources/preprocessors/preprocessor.py

```diff
@@ -1,129 +1,205 @@
-        no preprocessing files can be found.
+        Function to get the categorical columns that are within the supplied
+        annotation features of the imputing file.
+        :param dataset: pandas DataFrame
         """
-        error_message = 'No usable python files are found within the model directory!'
-        self.log.critical(error_message)
-        raise FileNotFoundError(error_message)
+        for feature in dataset.select_dtypes(include=["O"]).columns:
+            if feature not in self.exclude_features:
+                self.objects.append(feature)
+        self.log.debug(
+            'Converting the categorical columns: %s.',
+            ", ".join(self.objects)
+        )
 
-    def _load_correct_preprocessor(self):
+    def _process_objects(self, dataset: pd.DataFrame):
         """
-        Function to check the dynamically loaded preprocessors to match either the overrule argument or
-        the cadd version and genome build.
+        (If train) will create a dictionary telling the processor how many
+        categories are within a certain column.
+        If not train: Will look up each annotation feature from the impute file
+        within the columns of the datafile (either in full name or the column
+        starts with the feature from the impute file).
+        This dictionary is then passed to the actual processor.
+        :param dataset: unprocessed pandas DataFrame
+        :return: processed pandas DataFrame
         """
-        for preprocessor in self.preprocessors:
-            if self.overrule and preprocessor.name == self.overrule:
-                self.log.info('Overrule successful for: {} , located at: {}'.format(
-                    self.overrule, inspect.getfile(preprocessor.__class__)))
-                self.preprocessor = preprocessor
-                break
-            else:
-                module_cadd = preprocessor.supported_cadd_version
-                module_grch = preprocessor.supported_grch_build
-                if module_cadd == self.cadd_version and module_grch == self.grch_version:
-                    self.log.info('Preprocessing and model file successfully found: {} , Located at: {}'.format(
-                        preprocessor.name, inspect.getfile(preprocessor.__class__)))
-                    self.preprocessor = preprocessor
-                    break
-
-    def _check_preprocessor_is_applied(self):
-        if self.preprocessor is None:
-            if self.overrule:
-                error_message = 'No model data file found for overrule: {}'.format(
-                    self.overrule
-                )
-            else:
-                error_message = 'No model data file found for CADD version: {} and genome build: {}'.format(
-                    self.cadd_version,
-                    self.grch_version
+        annotation_feats_dict = {}
+        if self.train:
```

Comment:
Is this code really "specific to runmode train" or "what should be done when no `model_features` list is given"? If additional runmodes could result in more if/else clauses, perhaps a parent/child implementation would be better. If this code in the current state could be used by other modes, then it's not necessarily "train-specific" and more a "what to do if no `model_features` are given" and a `None` check on that would suffice (as `self.train` could be a bit misleading).